petstore/src/test/scala/io/finch/petstore/PetstorePetServiceSpec.scala

```diff
@@ -0,0 +1,218 @@
+package io.finch.petstore
+
+import com.twitter.finagle.{Service}
+import com.twitter.finagle.httpx.{FileElement, Request, RequestBuilder, Response}
+import com.twitter.io.{Buf, Reader}
+import com.twitter.util.{Await}
+import io.finch.test.ServiceSuite
+import org.scalatest.Matchers
+import org.scalatest.fixture.FlatSpec
+
+trait PetstorePetServiceSuite { this: FlatSpec with ServiceSuite with Matchers =>
+  def createService(): Service[Request, Response] = {
+    val db = new PetstoreDb()
+    val rover = Pet(None, "Rover", Nil, Some(Category(None, "dog")), Some(Seq(Tag(None, "puppy"),
+      Tag(None, "white"))), Some(Available))
+    db.addPet(rover)
+    db.addPet(Pet(None, "Sadaharu", Nil, Some(Category(None, "inugami")), Some(Nil), Some(Available)))
+    db.addPet(Pet(None, "Despereaux", Nil, Some(Category(None, "mouse")), Some(Nil), Some(Available)))
+    db.addPet(Pet(None, "Alexander", Nil, Some(Category(None, "mouse")), Some(Nil), Some(Pending)))
+    db.addPet(Pet(None, "Wilbur", Nil, Some(Category(None, "pig")), Some(Nil), Some(Adopted)))
+    db.addPet(Pet(None, "Cheshire Cat", Nil, Some(Category(None, "cat")), Some(Nil), Some(Available)))
+    db.addPet(Pet(None, "Crookshanks", Nil, Some(Category(None, "cat")), Some(Nil), Some(Available)))
+
+    val mouseCircusOrder: Order = Order(None, Some(4), Some(100), Some("2015-07-01T17:36:58.190Z"), Some(Placed),
+      Some(false))
+    db.addOrder(mouseCircusOrder)
+
+    val coraline: User = User(None, "coraline", Some("Coraline"), Some("Jones"), None, "becarefulwhatyouwishfor", None)
+    db.addUser(coraline)
+
+    endpoint.makeService(db)
+  }
+
+  //getPetEndpt test
+  "The PetstoreApp" should "return valid pets" in { f =>
+    val request = Request("/pet/1")
+    val result = f(request)
+    result.statusCode shouldBe 200
+   }
+
+  it should "fail to return invalid pets" in { f =>
+    val request = Request("/pet/100")
+    val result = Await.result(f.service(request))
+    result.statusCode shouldBe 404
+   }
+
+  //addPetEndpt test
+  it should "add valid pets" in { f =>
+    val request: Request = RequestBuilder()
+        .url("http://localhost:8080/pet").buildPost(
+          Buf.Utf8(s"""
+            |  {
+            |    "name": "Ell",
+            |    "photoUrls":[],
+            |    "category":{"name":"Wyverary"},
+            |    "tags":[{"name":"Wyvern"}, {"name":"Library"}],
+            |    "status":"pending"
+            |  }
+           """.stripMargin)
+        )
+    val result: Response = f(request)
+    result.statusCode shouldBe 200
+   }
+
+  it should "fail appropriately when adding invalid pets" in { f =>
+    val request: Request = RequestBuilder()
+        .url("http://localhost:8080/pet").buildPost(
+          Buf.Utf8(s"""
+            |  {
```

Comment:
Nitpicky but is there a reason this JSON block and the next are indented differently?